sklearn/metrics/tests/test_common.py

```diff
@@ -1588,6 +1588,33 @@ def test_regression_sample_weight_invariance(name):
     check_sample_weight_invariance(name, metric, y_true, y_pred)
 
 
+@pytest.mark.parametrize(
+    "name",
+    sorted(
+        set(ALL_METRICS).intersection(set(REGRESSION_METRICS))
+        - METRICS_WITHOUT_SAMPLE_WEIGHT
+    ),
+)
+def test_regression_invalid_sample_weight(name):
+    # Check that `sample_weight` with incorrect length raises error
+    n_samples = 50
+    random_state = check_random_state(0)
+    # regression
```

Comment:
I think we can remove this comment as the test already specifies we are testing for regression metrics.
```suggestion
```